qchem/pennylane_qchem/qchem/hf/integrals.py

```diff
@@ -692,15 +692,18 @@ def _boys(n, t):
 
     Args:
         n (float): order of the Boys function
-        t (float): exponent of the Boys function
+        t (array[float]): exponent of the Boys function
 
     Returns:
-        float: value of the Boys function
+        (array[float]): value of the Boys function
     """
-    if t == 0.0:
-        return 1 / (2 * n + 1)
-
-    return asp.special.gammainc(n + 0.5, t) * asp.special.gamma(n + 0.5) / (2 * t ** (n + 0.5))
+    return anp.where(
+        t == 0,
+        1 / (2 * n + 1),
+        asp.special.gammainc(n + 0.5, t + (t == 0))
+        * asp.special.gamma(n + 0.5)
+        / (2 * (t + (t == 0)) ** (n + 0.5)),
+    )  # (t == 0) is added to avoid divide by zero
```

Comment:
I see what's happening here, I just wonder if its needlessly complicated? Any reason why we can't iterate over t directly and use an `if-else` clause before casting back to `anp.array`? Does it break differentiability? 

This is just a suggestion, I think the line is fine, just a little hard to read while skimming the code. Feel free to leave it as is.

Reply:
We were doing such iteration previously which is correct but more costly because of the `for` loop and the `if` condition.